### ISSUE TITLE
Add a helper method to format/replace entities in text

### ISSUE THREAD

**Issue by xPaw** [linvi/tweetinvi#617]
This would make it much easier to pre-process a tweet before displaying it (e.g. expanding all urls, or coloring user mentions).

I can't think of a simple way of working with entity indices right now as each entity object is it's own type, and I would have to write some complicated logic to format each entity type correctly in the string. And since it's position based, entities would have to be sorted first to replace multiple entities correctly.

**Comment by linvi** [comment]
I am not sure I understand what you want exactly. Expanding urls can be done.

But how do you want Tweetinvi to `color` user mentions?

Please provide all the improvements that could be done and how you think the text should be improved.

**Comment by xPaw** [comment]
In my case, I want to color entities for IRC output, but this could also be HTML for example.

**Comment by linvi** [comment]
I don't think this should be the responsibility of a library like Tweetinvi to provide such formatting, but I keep this ticket open as an enhancement.

But this won't be done before at least version 3.2.

Note : Create a `ToString(OutputStyleEnum)` method for Tweets

**Comment by xPaw** [comment]
For reference, here's what I ended up doing: https://github.com/xPaw/WendySharp/blob/3c9160d52e843bd86bf505242da896695480f44b/WendySharp/LinkExpander/LinkExpander.cs#L260-L354

I put all known entities into list of entities. I had to create a new class for them because entities in tweetinvi don't have a common parent class.

Then I order that list by `entity.Start` from the end, iterate that list and do my transformations.

**Comment by linvi** [comment]
Thanks, I will take a look at it. If you could share Tweet ids where you can reproduce the issue it would be ideal.

Thanks

**Comment by xPaw** [comment]
Any tweet with emoji should do.

https://twitter.com/Emojipedia/status/961315134911270912

**Comment by linvi** [comment]
Thanks
